chore: add circleci to project

jon-nona opened the pull request:
# Pull Request

## Scope

adds circleci to the react playground

## Work Done

chore: add audit-ci step to circleci
chore: add circle ci validation steps to husky hooks
chore: update help.json with relevant information

## Steps to Test

- check ci passes
- check that you are happy with the circleci configuration

jon-nona committed:
chore: add circleci to project
chore: add audit-ci step to circleci
chore: add circle ci validation steps to husky hooks

neilrussell6 commented on .circleci/config.yml:
```diff
@@ -0,0 +1,97 @@
+
+base_node: &base_node
+  docker:
+  - image: circleci/node:8.10.0-browsers
```
please update all yaml array indetations to be:
```
  docker:
    - image: circleci/node:8.10.0-browsers
```
instead of
```
  docker:
  - image: circleci/node:8.10.0-browsers
```

jon-nona replied:
@neilrussell6 this should be resolved now

neilrussell6 commented on .circleci/config.yml:
```diff
@@ -0,0 +1,97 @@
+  name: 'Tests'
+  command: |
+    npm run test
+
```
remove extra newline

neilrussell6 commented on .circleci/config.yml:
```diff
@@ -0,0 +1,97 @@
+# --------------------------------
+
+jobs:
+  audit-linting-tests:
```
so ATM in our other projects we name this according to what flow it applies to / what the goal is
eg.
  pull-request-job
  deploy-production-job
etc
instead of specifically what it does
what are your thoughts on this?
what would be the benefit of this approach?
I find this more descriptive:
```
  build:
    jobs:
      - pull-request-job:
          filters:
            branches:
              ignore:
                - master
                - develop
      - deploy-staging-job:
          context: Development
          filters:
            branches:
              only: develop
```
ie.
run general PR job on any branches except master or develop
and run the deploy to staging job on develop

is there a way to compose jobs that would make it useful to name them according to their steps?

neilrussell6 commented on help.json:
```diff
@@ -1,4 +1,6 @@
 {
+  "audit": "audit npm dependencies for critical vulnerabilities",
```
can we call this:
```
npm:audit
```
?

jon-nona committed:
chore: fix indentations on circleci yml
chore: change npm command names and help description
chore: update prettier config to use single quotes where possible.
